### ISSUE TITLE
Brave crashes when trying to view saved password

### ISSUE THREAD

**Issue by gabrielrozn** [brave/brave-browser#21231]
## Description 

Brave crashes when trying to view saved password

## Steps to Reproduce

   1. Enter password management page ->  brave://settings/passwords
   2. Choose to view or copy one of the saved passwords.
   3. Windows login ID is requested, after filling the windows password and pressing enter, Brave crashed.

## Actual result:

Brave is force closed.

## Expected result:
Password can be viewed and/or modified

## Reproduces how often: 

Every time

## Brave version (brave://version info)

Brave | 1.35.103 Chromium: 98.0.4758.102 (Official Build) (64-bit)
Revision | 273bf7ac8c909cde36982d27f66f3c70846a3718-refs/branch-heads/4758@{#1151}
OS | Windows 7 Service Pack 1 (Build 7601.23964)

## Version/Channel Information:

- Can you reproduce this issue with the current release? 
 Yes
- Can you reproduce this issue with the beta channel? 
didn't try
- Can you reproduce this issue with the nightly channel? 
didn't try
## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
didn't try
- Does the issue resolve itself when disabling Brave Rewards? 
didn't try
- Is the issue reproducible on the latest version of Chrome? 
YES
## Miscellaneous Information:

Error Report
Source
Brave Browser

Summary
Stopped working

Date
‎22/‎02/‎2022 11:49

Status
Not reported

Description
Faulting Application Path:	C:\Program Files\BraveSoftware\Brave-Browser\Application\brave.exe

Problem signature
Problem Event Name:	APPCRASH
Application Name:	brave.exe
Application Version:	98.1.35.103
Application Timestamp:	6206cd90
Fault Module Name:	StackHash_423a
Fault Module Version:	6.1.7601.24441
Fault Module Timestamp:	5cb9356e
Exception Code:	c0000374
Exception Offset:	00000000000bf302
OS Version:	6.1.7601.2.1.0.256.1
Locale ID:	1046
Additional Information 1:	423a
Additional Information 2:	423adab2099b2868015cb9460891eb88
Additional Information 3:	bef7
Additional Information 4:	bef7ec23ba7820d0d0a9e6f63d1659a3

Files that help describe the problem
AppCompat.txt
WERInternalMetadata.xml
memory.hdmp
minidump.mdmp

[crashdump.zip](https://github.com/brave/brave-browser/files/8119192/crashdump.zip)

**Comment by rebron** [comment]
cc: @btlechowski

**Comment by kjozwiak** [comment]
Seems to be working on `Win 11 x64` so this might be an issue with `Win 7 x64`. Basically tried the following using the following build(s):

```
Brave | 1.38.9 Chromium: 99.0.4844.45 (Official Build) nightly (64-bit)
-- | --
Revision | edbc0b8343c7b10fddb0e1b4efb280b0f6e38cab-refs/branch-heads/4844@{#788}
OS | Windows 11 Version 21H2 (Build 22000.527)
```

* opened `brave://settings/passwords`
* clicked on the `Show password` button next to a random password that I saved
* entered my `Windows` PIN and clicked `OK`
* password that I clicked on became visible
* ran through several other passwords by clicking on `Show password` and they all appeared without issues/Brave crashing

![image](https://user-images.githubusercontent.com/2602313/156240973-32310761-6456-41b0-a7bd-6cf399eb2b13.png)

I also checked `Win 8.1 x64` via a VM and that seems to be working as well using the following build(s):

```
Brave | 1.35.103 Chromium: 98.0.4758.102 (Official Build) (64-bit)
--- | ---
Revision | 273bf7ac8c909cde36982d27f66f3c70846a3718-refs/branch-heads/4758@{#1151}
OS | Windows 8.1 (Build 9600.20269)
```

Basically used the same STR/Cases mentioned above and once I went through the `Win 8.1` authentication prompt, I was able to view the password within `brave://settings/passwords` without issues.

<img width="1220" alt="Screen Shot 2022-03-01 at 3 50 30 PM" src="https://user-images.githubusercontent.com/2602313/156247139-39455f86-fff9-424e-8996-f334235d7f75.png">

Lets see what @btlechowski sees on `Win 7` 👍
